Elf/__init__.py

```diff
@@ -517,10 +517,14 @@ def getNotes(self):
                 continue
 
             notebytes =  self.readAtOffset(sec.sh_offset, sec.sh_size)
-
             offset = 0
-            while offset < len(notebytes):
+            notebyteslen = len(notebytes)
+            while offset < notebyteslen:
                 note = vs_elf.ElfNote()
+                if notebyteslen - offset < len(note):
+                    print ("\nNOTES section length mismatch!\n\t%s\n\tSection Bytes: %s\n\tStranded bytes: %s\n" % (sec, repr(notebytes), repr(notebytes[offset:])))
```

Comment:
i wouldn't object to williballenthin's suggestion for using the "logging" module for non-performance things, particularly non-direct viv things.  Verbose mode could by used for both log level, and for engaging the debug listeners for all modules.

however, for that method to be successful, viv and associated modules should have a set naming pattern and usage strategy for logging so that there's no guesswork.  without one, logs could be issued for some way out oddball logging target that viv would never know to listen for.  

your call.  we can hand in an optional logging routine (but all non-viv modules would need it), we can use the logging module, or we can do something else.